modules/language-guide/pages/sharing.adoc

```diff
@@ -1,12 +1,228 @@
+
+For simplicity, we assume a very simple type for `notify` above.  In
+general, this `notify` function would accept relevant notification
+data and perhaps return some new status of the subscriber to the
+publisher, e.g., changing its subscription settings based on the
+notification data.
+
+
+=== Publisher side
+
+The publisher side of the code stores an array of subscribers, where
+we assume (for simplicity) that each subscriber only subscribes
+themselves once, via `subscribe`.
+
+[source,motoko]
+----
+actor Publisher {
+    var subs: [Subscriber] = [];
+
+    public func subscribe(sub: Subscriber) {
+        subs := Array.append<Subscriber>(subs, [sub]);
+    };
+
+    public func publish() {
+        for (sub in subs.vals()) {
+          sub.notify();
+        };
+    };
+};
+----
+
+Later, when some unspecified external agent invokes `publish`, all of
+the subscribers receive the `notify` message, as in the `Subscriber`
+type given above.
+
+=== Subscriber side
+
+In the simplest case, the subscriber side has three methods:
+1. initially, subscribe to notifications from the publisher (`init`),
+2. be notified as one of the subscribed actors (`notify`, as in the `Subscriber` type given above), and
+1. permit queries to the accumulated state, in this case just a counter of notifications (`get`).
+
+The code below implements each of these:
+
+[source,motoko]
+----
+import Publisher "canister:pub";
+
+actor Subscriber {
+    var count: Nat = 0;
+    public func init() {
+        Publisher.subscribe(Subscriber);
+    };
+    public func notify() {
+        count += 1;
+    };
+    public func get() : async Nat {
+        count
+    };
+}
+----
+
+The actor assumes, but does not enforce, that its `init` function is
+only ever called once.  In it, the `Subscriber` actor passes a
+reference to itself, of type `actor { notify : () -> () };` (locally
+called `Subscriber` above).
+
+If called more than once, the actor will subscribe itself multiple
+times, and will receive multiple (duplicate) notifications from the
+publisher.  This fragility is the consequence of the basic
+publiisher-subscriber design we show above.  With more care, a more
+advanced publisher actor could check for duplicate subscriber actors
+and ignore them, for instance.
+
+== Sharing functions among actors
+
+In {proglang}, a `shared` actor function can be sent in a message to
+another actor, and then later called by that actor, or another one.
+
+The code shown above has been simplified for illustrative purposes.
+The full version offers additional features to the
+publisher-subscriber relationship, and uses shared functions to make
+this relationship more flexible.
+
+For instance, the notification function is _always_ designated as
+`notify`.  A more flexible design would only fix the type of `notify`,
+and permit the subscriber to choose any of its `shared` functions,
+specified in a `subscribe` message in place of (just) the actor that
+is subscribing.
```

Comment:
```suggestion
For instance, the notification function is an actor member named 
`notify`.  A more flexible design would only fix the signature of some notification function,
and permit the subscriber to choose any of its `shared` functions that match the signature, regardless of name,
specified in a `subscribe` message as an individual function rather than some containing actor.
This would allow an actor to register two different notification functions with the same or even separate publishers, for example.
```